sockeye/beam_search.py

```diff
@@ -585,10 +604,31 @@ def forward(self,
         # TODO: check for disabled predicted output length
 
         t = 1
+        if target_prefix_factors is not None and C.TARGET_FACTOR_SHIFT:
```

Comment:
Thanks. 
I created a utils.shift_prefix_factors function for this and call it from _get_inference_input() to avoid calling it in different places (between beamsearch and greedysearch).

`I target_prefix_factors = utils.shift_prefix_factors(target_prefix_factors) if target_prefix_factors is not None and C.TARGET_FACTOR_SHIFT else target_prefix_factors
`

Here is also the comment to explain while we need to shift prefix factors:

`# During inference, if C.TARGET_FACTOR_SHIFT is True, predicting target_factors are left-shifted (see _unshift_target_factors function()) so that they re-align with the words. With that, target_prefix_factors need to be also right-shifted here if C.TARGET_FACTOR_SHIFT is True so that when they are shifted back later they would align with words.
`